src/game/mcts.rs

```diff
@@ -23,14 +34,13 @@ struct MCTSNode<Position: GamePosition> {
     simulations_n: u32,
 
     /// This is the variable w from UCT formula
-    /// Float because could be half for games with ties
     score_w: f32,
 }
 
-impl<Position: GamePosition> MCTSNode<Position> {
-    pub fn from_position(position: Position, init_score: f32) -> Self {
+impl<Move: GameMove> MCTSEdge<Move> {
```

Comment:
General question about this commit - why is it better to have the info on the edges?
1. If the search "tree" really is a tree, then it makes no real difference, right?
The only difference is that the current/old implementation is not intuitive about the meaning of a node's `score` relative to that node's "player to play"? 

2. If we allow the search tree to be a DAG:
* Remind me - is this really a possible future scenario?
* In this scenario, would we want the info to be on edges or nodes? Why? (My first intuition is nodes)

Reply:
The change is only for intuitive benefits, but it does complicate the code a little bit, I think it worth it.
DAG will complicate backpropagation, and will require global table of pos->node, which might be worth it
I guess in such case we will prefer data on nodes

Reply:
Sababa